var.mk

```diff
@@ -14,14 +14,29 @@ PROGRAMATICA_SRC_FILE ?= \
 # The local file gets tried first, and if not usable the remote on gets fetched.
 # If both are unset or set to an empty string, programatica support is skipped.
 
-GHCVERSION := $(shell ghc --numeric-version)
+OSNAME := $(shell uname -s)
+OSVERS := $(shell uname -v 2>/dev/null)
+
+STACK := stack
+STACK_EXEC := $(STACK) exec --
+
+ifeq "$(OSNAME)" "Darwin"
+    STACK_DEPENDENCIES_FLAGS := --flag gtk:have-quartz-gtk
+else
+    STACK_DEPENDENCIES_FLAGS :=
+endif
+
+# Dummy variables to install GHC and install the Haskell-dependencies.  These
+# must be called before ghc or ghc-pkg is called because the packages need to
+# be installed before they are checked.
+STACK_SETUP := $(shell $(STACK) setup > /dev/null)
+STACK_INSTALL_DEPENDENCIES := $(shell $(STACK) build --only-dependencies $(STACK_DEPENDENCIES_FLAGS) > /dev/null)
+
```

Comment:
Not sure, what it really does. If it is really only for setting up/building stack, than a simple target for convenience like this would be IMHO sufficient:
```
stack:
    $(STACK) setup
    $(STACK) build --only-dependencies $(STACK_DEPENDENCIES_FLAGS)
```

So if the developer hasn't set it up yet, he just needs to 'make stack' once and than forget it. As said, building hets should NOT be interweaved with setting up the tool chain, that's it.
However, because you made the setup part of the Makefile, you also need to include a receipt to remove the so installed stuff cleanly, i.e. w/o removing anything, which is not hets related/shared by other build stuff. So you hopefully see, that pulling in the stack setup is not very wise and opens a plethora of other problems - just a waste of time to fix it. The better thing one would do is to document 'stack' as an [optional] requirement and possibly add some links, which point to articles, how to setup this stuff.